sentry/src/main/java/io/sentry/TracesSampler.java

```diff
@@ -10,14 +11,14 @@ final class TracesSampler {
   private static final @NotNull Double DEFAULT_TRACES_SAMPLE_RATE = 1.0;
 
   private final @NotNull SentryOptions options;
-  private final @NotNull Random random;
+  private final @Nullable Random random;
 
   public TracesSampler(final @NotNull SentryOptions options) {
-    this(Objects.requireNonNull(options, "options are required"), new Random());
+    this(Objects.requireNonNull(options, "options are required"), null);
```

Comment:
```suggestion
    this(Objects.requireNonNull(options, "options are required"), SentryRandom.current());
```
Can't we just pass current random instead of nullifying the variable?

Reply:
nvm, there's the problem for different threads, right?

Reply:
Yeah, you're supposed to retrieve an instance, when you use it and not reference it across threads as that would defeat the purpose.